docs/05-rfcs/2023-01-11-sdk-spec.md

```diff
@@ -0,0 +1,1070 @@
+
+  /**
+   * Run an inflight whenever the schedule triggers.
+   */
+  on_tick(fn: inflight () => void, opts: ScheduleOnTickProps?): void;
+}
+```
+
+Example:
+
+```ts
+// Run a function every hour.
+let every_hour = new cloud.Schedule.fromRate(1 hour); // implicit scope and id?
+
+every_hour.on_tick(inflight () => {
+  // ...
+});
+```
+
+Future extensions: inflight `next_tick(): Duration` method?
+
+## Website
+
+The website resource represents a CDN-backed website.
+It is useful for hosting static content, such as a blog or a single-page application.
+
+```ts
+struct WebsiteProps {
+  /**
+   * Local path to the website's static files, relative to the Wing source file.
+   * @example "./dist"
+   */
+  path: str;
+
+  /**
+   * The website's domain name.
+   * @example "example.com"
+   */
+  domain: str;
+}
+
+resource Website {
+  init(props: WebsiteProps);
+
+  /**
+   * The website's url.
+   */
+  url: str;
+}
+```
+
+Future extensions: domain and certificate props? support for edge functions?
+
+## Api
+
+The Api resource represents an API Gateway that can be used to manage HTTP routes and run functions in response to requests.
+
+```ts
+struct ApiProps {
+  cors: ApiCorsProps?;
+}
+
+resource Api {
+  init(props: ApiProps = {});
+
+  /**
+   * The api's base url.
+   */
+  url: str;
+
+  /**
+   * Run an inflight whenever a GET request is made to the specified route.
+   */
+  get(route: str, fn: inflight (req: ApiRequest) => ApiResponse, opts: cloud.ApiOnGetProps?): void;
+
+  /**
+   * Run an inflight whenever a POST request is made to the specified route.
+   */
+  post(route: str, fn: inflight (req: ApiRequest) => ApiResponse, opts: cloud.ApiOnPostProps?): void;
+
+  /**
+   * Run an inflight whenever a PUT request is made to the specified route.
+   */
+  put(route: str, fn: inflight (req: ApiRequest) => ApiResponse, opts: cloud.ApiOnPutProps?): void;
+
+  /**
+   * Run an inflight whenever a DELETE request is made to the specified route.
+   */
+  delete(route: str, fn: inflight (req: ApiRequest) => ApiResponse, opts: cloud.ApiOnDeleteProps?): void;
+
+  /**
+   * Run an inflight whenever a PATCH request is made to the specified route.
+   */
+  patch(route: str, fn: inflight (req: ApiRequest) => ApiResponse, opts: cloud.ApiOnPatchProps?): void;
+
+  /**
+   * Run an inflight whenever any request is made to the specified route.
+   */
+  add_route(route: str, fn: inflight (req: ApiRequest) => ApiResponse, opts: cloud.ApiOnRequestProps?): void;
+
+  /**
+   * Make a request to the specified route. Throws if the route hasn't been
+   * defined.
+   */
+  inflight request(route: str, method: HttpMethod, body: Serializable): Serializable;
+}
+
+struct ApiOnGetProps { /* elided */ }
+struct ApiOnPostProps { /* elided */ }
+struct ApiOnPutProps { /* elided */ }
+struct ApiOnDeleteProps { /* elided */ }
+struct ApiOnPatchProps { /* elided */ }
+struct ApiOnRequestProps { /* elided */ }
+
+struct ApiCorsProps {
```

Comment:
Not sure about this name, what's "cors"?

Reply:
https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS

Reply:
Yeah - I think anyone that needs to use this option will be familiar with the acronym. 

BTW we should see if it's possible to make this work @yoav-steinberg:

```js
// wing

// OK
new Api(
  cors: ApiCorsProps {
    origins: ["https://example.com"],
  }
}

// better
new Api(
  cors: {
    origins: ["https://example.com"],
  }
}
```